### PR TITLE
Added product and owner details in Service area's page

### PR DESCRIPTION
**Pull request by Sandhya1874**

### CODE REVIEW

File: assets/js/serviceAreas.js [hunk 1]
```diff
@@ -6,6 +6,26 @@ jQuery(function () {
...
+    {
+      data: 'attributes.owner',
+      createdCell: function (td, _cellData, rowData) {
+        $(td).html(`${rowData.attributes.owner}`)
+      },
+    },
+    {
+      data: 'attributes.products',
+      createdCell: function (td, _cellData, rowData) {
+        for (const key in rowData.attributes.products) {
```
Comment (psoleckimoj) [inline assets/js/serviceAreas.js:18]: to do this in a more functional way use 

```rowData.attributes.products.forEach(product => {
...
}```

Comment (psoleckimoj) [reply assets/js/serviceAreas.js:18]: Actually you're got a for loop around a foreach on the same data?

Comment (psoleckimoj) [reply assets/js/serviceAreas.js:18]: Actually I can't work out what this is doing?

File: assets/js/productSets.js [hunk 2]
```diff
@@ -12,6 +12,20 @@ jQuery(function () {
...
+      data: 'attributes.products',
+      createdCell: function (td, _cellData, rowData) {
+        const header = `<details class="govuk-details"><summary class="govuk-details__summary"><span class="govuk-details__summary-text">Product List</span></summary>`
+        const products = rowData.attributes.products.data
+          .map(product => {
+            return `<li><a href="/products/${product.attributes.slug}" data-test="product-${product.id}">${product.attributes.name}</a></li>`
+          })
+          .join('')
+
+        if (products) $(td).html(`${header}${products}`)
```
Comment (psoleckimoj) [inline assets/js/productSets.js:25]: Just for our linting in the main code, it's better to do:

```
if () {
} else {
}
```

File: assets/js/productSets.js [hunk 3]
```diff
@@ -12,6 +12,20 @@ jQuery(function () {
         $(td).html(`<a href="/product-sets/${rowData.id}">${rowData.attributes.name}</a>`)
       },
     },
+    {
+      data: 'attributes.products',
+      createdCell: function (td, _cellData, rowData) {
+        const header = `<details class="govuk-details"><summary class="govuk-details__summary"><span class="govuk-details__summary-text">Product List</span></summary>`
+        const products = rowData.attributes.products.data
```
Comment (psoleckimoj) [inline assets/js/productSets.js:19]: Looks much better 🙂

File: assets/js/serviceAreas.js [hunk 4]
```diff
@@ -15,14 +15,14 @@ jQuery(function () {
     {
       data: 'attributes.products',
       createdCell: function (td, _cellData, rowData) {
-        for (const key in rowData.attributes.products) {
-          var pList = ''
-          pList = `<details class="govuk-details"><summary class="govuk-details__summary"><span class="govuk-details__summary-text">Product List</span></summary>`
-          rowData.attributes.products[key].forEach(product => {
-            pList += `<li><a href="/products/${product.attributes.slug}" data-test="product-${product.id}">${product.attributes.name}</a></li>`
+        const header = `<details class="govuk-details"><summary class="govuk-details__summary"><span class="govuk-details__summary-text">Product List</span></summary>`
+        const products = rowData.attributes.products.data
```
Comment (psoleckimoj) [inline assets/js/serviceAreas.js:19]: Could pull this out into a common util function as it's been used across pages?

File: server/data/strapiApiClient.ts [hunk 5]
```diff
@@ -119,18 +119,25 @@ export default class StrapiApiClient {
...
   }
 
-  async getNamespaces({ withRdsInstances = false }: { withRdsInstances?: boolean }): Promise<NamespaceListResponse> {
-    const populate = []
-
-    if (withRdsInstances) {
-      populate.push('rds_instance')
-    }
-
+  // async getNamespaces({ withRdsInstances = false }: { withRdsInstances?: boolean }): Promise<NamespaceListResponse> {
```
Comment (psoleckimoj) [inline server/data/strapiApiClient.ts:122]: Best not to checkin commented out code, either remove or uncomment.

File: server/views/pages/productSets.njk [hunk 6]
```diff
@@ -31,4 +33,4 @@
   <script src="/assets/moj/all.js"></script>
   <script src="/assets/js/common.js"></script>
   <script src="/assets/js/productSets.js"></script>
-{% endblock %}
+{% endblock %}
```
Comment (psoleckimoj) [inline server/views/pages/productSets.njk:36]: Missing trailing newline.

File: assets/js/teams.js [hunk 7]
```diff
@@ -18,20 +18,20 @@ jQuery(function () {
     {
       data: 'attributes.products',
       createdCell: function (td, _cellData, rowData) {
-        for (const key in rowData.attributes.products) {
-          var pList = ''
-          rowData.attributes.products[key].forEach(product => {
-            pList += `<li><a href="/products/${product.attributes.slug}" data-test="product-${product.id}">${product.attributes.name}</a></li>`
+        const products = rowData.attributes.products.data
+          .map(product => {
+            return `<li><a href="/products/${product.attributes.slug}" data-test="product-${product.id}">${product.attributes.name}</a></li>`
```
Comment (psoleckimoj) [inline assets/js/teams.js:23]: Just for learning, with arrow functions you can rewrite this as below, if you're just doing a return you can usually skip it and the brackets:

```
(product => `<li><a href="/products/${product.attributes.slug}" data-test="product-${product.id}">${product.attributes.name}</a></li>`)
```

**Review by psoleckimoj — DISMISSED** [review]
LGTM, a couple of tidy ups in my comments but can always do after in another PR.

File: assets/js/teams.js (hunk 7 above)
Comment (Sandhya1874) [reply assets/js/teams.js]: Modified as suggested

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**